app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class SessionsController < ApplicationController
+
+  before_action :user, only: [:create]
+
+  def new
+    if current_user
+      redirect_to user_path(current_user), alert: t(".alert")
+    end
+  end
+
+  def create
+    if user&.authenticate(params[:session][:password])
+      session[:user_id] = user.id
+      redirect_to user_path(user), notice: t(".notice")
+    else
+      flash[:alert] = t(".alert")
+      render :new
+    end
+  end
+
+  def destroy
+    session.clear
+    redirect_to root_path, notice: t(".notice")
+  end
+
+  private
+
+  def user
+    User.find_by(email: params[:session][:email])
```

Comment:
What do you think about memoizing this query? (i.e. `@user ||= User.find_by...`)

Reply:
Sounds good, again stuck in a rut with patterns I am used to!